### ISSUE TITLE
pyfetch await json or string response from failed request raises OSError instead of returning response payload

### ISSUE THREAD

**Issue by owenlamont** [pyodide/pyodide#3974]
## 🐛 Bug

When a request made with pyfetch returns a not okay status code attempting to retrieve the body of the response as a string or json raises an OSError instead of returning. This is a follow up to issue [3768 ](https://github.com/pyodide/pyodide/issues/3768) where I misunderstood the problem.

### To Reproduce

I did this on pyscript.com but can run this locally anywhere you can get a CORS valid URL that returns a not okay status code

**index.html**

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <link rel="stylesheet" href="https://pyscript.net/releases/2023.03.1/pyscript.css" />
    <script defer src="https://pyscript.net/releases/2023.03.1/pyscript.js"></script>
</head>
<body>
    <py-config src="./pyscript.toml"></py-config>
    <py-script src="./main.py"></py-script>
</body>
</html>
```

**main.py**

```python
import asyncio
import json
import pyodide.http as http

async def attempt_request():
    response = await http.pyfetch(url="/",
                                  headers={"Content-Type": "application/json"},
                                  method="POST",
                                  body=json.dumps({"Take": "this"}))
    print(response.status)
    json_response = await response.json()
    print(json_response)
    

asyncio.ensure_future(attempt_request())
```

**pyscript.toml**

```toml
packages = [
    "pyodide-http"
]

[[interpreters]]
src = "https://cdn.jsdelivr.net/pyodide/v0.23.3/full/pyodide.js"
name = "pyodide-0.23.3"
lang = "python"
```

### Expected behavior

I expect no exception to be thrown and the payload of the response to be printed. In this case I expected to see:

```json
{
  "detail": "Method Not Allowed"
}
```

### Environment

- Pyodide Version: 0.23.3
- Browser version: Vivaldi 6.1.3035.111 (Stable channel) (64-bit) 

### Additional context

![image](https://github.com/pyodide/pyodide/assets/12672027/ebfd5a41-9758-4db6-9c4b-2bf4499efaf8)
![image](https://github.com/pyodide/pyodide/assets/12672027/5cb42a61-be3a-4e62-b044-08c6079d5bf6)

**Comment by ryanking13** [comment]
Thanks for the report @owenlamont! Indeed it looks like we are raising an exception if the status_code >= 400 when accessing the response body.

https://github.com/pyodide/pyodide/blob/50b5e1ccb6863507c46f832adb743be0917b4dc1/src/py/pyodide/http.py#L128-L134

https://github.com/pyodide/pyodide/blob/50b5e1ccb6863507c46f832adb743be0917b4dc1/src/py/pyodide/http.py#L154-L166

I think the current behavior is wrong. Regardless of the response status code, one should be able to read the response body. Are you interested in opening a PR for this?

**Comment by owenlamont** [comment]
Seems like a straight forward change from those code snippets. Yes I can have an attempt at that.

**Comment by ryanking13** [comment]
Closing as resolved by #3986. Thanks for reporting and fixing this @owenlamont!
